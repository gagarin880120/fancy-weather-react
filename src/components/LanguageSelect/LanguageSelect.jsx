import React from 'react';
import styles from './LanguageSelect.module.css';

export default function LanguageSelect() {
  return (
    <select name="language" className={styles.select}>
      <option value="en" selected>en</option>
      <option value="ru">ru</option>
    </select>
  );
}
