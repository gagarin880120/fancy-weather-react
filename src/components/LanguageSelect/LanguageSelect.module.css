.select {
  background-color: transparent;
  color: #fff;
  outline: 0;
  height: 35px;
  cursor: pointer;
  font-size: 22px;
  font-family: Nunito;
  font-weight: bold;
  border-radius: 5px;
  text-transform: uppercase;
  border: 2px solid #fff;
}

.option {
  color:#000;
}

.select:active {
  transform: translateY(2px);
}
