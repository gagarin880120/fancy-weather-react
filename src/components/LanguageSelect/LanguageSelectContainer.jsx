import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import LanguageSelect from './LanguageSelect';
import { setLanguage } from '../../redux/actions';

export function LanguageSelectContainer({ onLanguageChange }) {
  return (
    <LanguageSelect
      onLanguageChange={onLanguageChange}
    />
  );
}

export const mapDispatchToProps = (dispatch) => ({
  onLanguageChange(lang) {
    dispatch(setLanguage(lang));
  },
});

LanguageSelectContainer.propTypes = {
  onLanguageChange: PropTypes.func,
};

LanguageSelectContainer.defaultProps = {
  onLanguageChange: () => {},
};

export default connect(null, mapDispatchToProps)(LanguageSelectContainer);
