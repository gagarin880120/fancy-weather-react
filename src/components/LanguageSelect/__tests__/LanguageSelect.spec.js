import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer, { act } from 'react-test-renderer';
import LanguageSelect from '../LanguageSelect';

describe('LanguageSelect', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<LanguageSelect />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });

    describe('Event', () => {
      const onLanguageChange = jest.fn();

      const testRenderer = TestRenderer.create(
        <LanguageSelect
          onLanguageChange={onLanguageChange}
        />
      );
      test('onChange should call onLanguageChange function', () => {
        const select = testRenderer.root.findByType('select');

        act(() => {
          select.props.onChange({target: { value: '28' }});
        })

        expect(onLanguageChange).toHaveBeenCalled();
      });
    });
  });
});
