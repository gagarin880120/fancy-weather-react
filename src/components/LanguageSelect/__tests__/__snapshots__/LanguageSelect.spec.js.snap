// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelect component should render without crashing 1`] = `
<select
  className="select"
  name="language"
  onChange={[Function]}
>
  <option
    className="option"
    value="en"
  >
    en
  </option>
  <option
    className="option"
    value="ru"
  >
    ru
  </option>
</select>
`;
